tests/switchless/host/host.c

```diff
@@ -12,15 +12,46 @@
 #if _MSC_VER
 #include <Windows.h>
 #endif
+#include "../../../host/hostthread.h"
 #include "../../../host/strings.h"
 #include "switchless_u.h"
 
+// Increase this number to have a meaningful performance measurement
+#define NUM_OCALLS (100000)
+
 #define STRING_LEN 100
 
 #if _MSC_VER
 static double frequency;
 #endif
 
+static int thread_create(oe_thread_t* thread, void* (*func)(void*), void* arg)
+{
+#if __GNUC__
```

Comment:
The #if tests look to me to be testing the wrong thing.   This looks more about whether it's Linux/posix or windows, rather than which compiler you're using.   So things like __attribute__ (which isn't used here) should be inside such checks, but Linux vs windows tests should not be based on any assumption about the compiler.

Reply:
Good point. I changed code to use __linux__/_WIN32 instead.

Additionally, I switched to using oe_thread_create/oe_thread_join which removes all this logic in the test. I also removed it from tests/switchless_threads/host/host.c where I had borrowed it from.